main/src/ca/uwaterloo/flix/api/lsp/Index.scala

```diff
@@ -178,6 +178,12 @@ case class Index(m: Map[(String, Int), List[Entity]],
     }
   }
 
+  /**
+   * Returns all entities in the document at the given `uri`.
+   */
+  def query(uri: String): List[Entity] =
+    m.view.filterKeys { case (s, _) => s == uri }.values.flatten.toList
```

Comment:
This is probably not very efficient with the use of `flatten` and `toList`. I guess it totally depends on: How many entities will there be? Maybe its not a practical problem.